File: Sliding window/LET_no487_Max Consecutive Ones II.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        prev = -1
        cur = res = 0
        for num in nums:
            if num == 0:
                prev, cur = cur, 0
            else:
                cur += 1
            
            res = max(res, prev + cur + 1)
        
        return res

File: Sliding window/LET_no487_Max Consecutive Ones II.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int prev = -1;
        int cur = 0;
        int res = 0;
        
        for (int &num : nums) {
            if (num == 0) {
                prev = cur;
                cur = 0;
            }
            else 
                cur++;
            
            res = max(res, prev + cur + 1);
        }
        
        return res;
    }
};